crates/storage/provider/src/providers/database/mod.rs

```diff
@@ -144,6 +149,12 @@ impl<N: NodeTypesWithDB> StorageSettingsCache for ProviderFactory<N> {
     }
 }
 
+impl<N: NodeTypesWithDB> RocksDBProviderFactory for ProviderFactory<N> {
```

Comment:
we should have an impl like this but for `BlockchainProvider` in crates/storage/provider/src/providers/blockchain_provider.rs:
```rust
  impl<N: ProviderNodeTypes> RocksDBProviderFactory for BlockchainProvider<N> {
      fn rocksdb_provider(&self) -> &RocksDBProvider {
          self.database.rocksdb_provider()
      }
  }
```
this follows the same delegation pattern used by `StaticFileProviderFactory`